### PR TITLE
vinyl: implement iterator pagination

### LINKED ISSUE #7634: Vinyl tree select pagination
Implement options `after` and `fetch_pos` in select method for vinyl tree. If it's specified it must contain a tuple from which selection must continue or its position. More strictly:

- For forward iteration (EQ, GE, GT, ALL) select must internally skip all tuples less-or-equal-to tuple given in after in terms of that index.
- For reverse iteration (REQ, LE, LT) select must internally skip all tuples greater-or-equal-to tuple given in after in terms of that index.

The feature must work with multikey indexes.

Part of #7639

### PR DESCRIPTION
**Pull request by locker**
Pagination for vinyl works exactly like in memtx. See the commit log for implementation details.

Closes #7634

### CODE REVIEW

File: test/engine-luatest/pagination_test.lua [hunk 1]
```diff
@@ -1,7 +1,7 @@
 local server = require('test.luatest_helpers.server')
 local t = require('luatest')
 local tree_g = t.group('Tree index tests',
-    t.helpers.matrix{engine={'memtx'}, disable_ffi={true, false}})
+    t.helpers.matrix{engine={'memtx', 'vinyl'}, disable_ffi={true, false}})
```
Comment (drewdzzz) [inline test/engine-luatest/pagination_test.lua:4]: We don't need to disable ffi now (for `vinyl`, at least)

**Review by drewdzzz — APPROVED** [review]
LGTM, but please address my comment

File: test/engine-luatest/pagination_test.lua (hunk 1 above)
Comment (locker) [reply test/engine-luatest/pagination_test.lua]: Right. Removed vinyl+ffi from the test plan:

<details><summary>diff</summary>

```diff
diff --git a/test/engine-luatest/pagination_test.lua b/test/engine-luatest/pagination_test.lua
index 553d3a2380ba..46e12865f554 100644
--- a/test/engine-luatest/pagination_test.lua
+++ b/test/engine-luatest/pagination_test.lua
@@ -1,7 +1,11 @@
 local server = require('test.luatest_helpers.server')
 local t = require('luatest')
-local tree_g = t.group('Tree index tests',
-    t.helpers.matrix{engine={'memtx', 'vinyl'}, disable_ffi={true, false}})
+local tree_g = t.group('Tree index tests', {
+    {engine = 'memtx', disable_ffi = true},
+    {engine = 'memtx', disable_ffi = false},
+    -- Vinyl doesn't use FFI.
+    {engine = 'vinyl', disable_ffi = false},
+})
 
 -- TODO(gh-7739): Enable Lua JIT on aarch64.
 local function start_server(server)
```

</details>

Comment (drewdzzz) [reply test/engine-luatest/pagination_test.lua]: Nice. If I am the only reviewer, please hang `full-ci` label.

**Pull request merged**